### ISSUE TITLE
Capture area of a monitor

### ISSUE THREAD

**Issue by michiproep** [Microsoft/Windows.UI.Composition-Win32-Samples#119]
Is there a way to capture only an area of the desktop or monitor by a given Rectangle?

**Comment by robmikh** [comment]
No, right now you need to capture the entire monitor and copy out the portion you're interested in using CopySubresourceRegion.

**Comment by michiproep** [comment]
do you have a short code snippet for this?

**Comment by robmikh** [comment]
I don't have a self-contained snippet to share with you, but the idea is straight forward. Let's say that you have two textures, texture A comes from the capture API and texture B is the texture that will receive the pixels you're interested in. You would call [`CopySubresourceRegion`](https://learn.microsoft.com/en-us/windows/win32/api/d3d11/nf-d3d11-id3d11devicecontext-copysubresourceregion) on the `ID3D11DeviceContext` and supply the area of interest from texture A and specify that it will be copied into texture B.

[`robmikh/GifSnip`](https://github.com/robmikh/GifSnip/blob/01cdfd5a028f0066927165b18e38b5f9278e84ad/GifSnip/FrameCompositor.cpp#L40-L64) does this very thing to copy a portion of the screen to an output texture used to encode a gif.

**Comment by Marcets423** [comment]
Here are some snippets of the relevant code I used. It's just the relevant lines of my code, It won't run.



```
Member variable:
D3D11_BOX m_roiWindow; //Region of interest rect
ID3D11Texture2D* m_stagingTexture= NULL;
//Set Function
SetROIWindow(float x, float y, float width, float height) {
    m_roiWindow.left   = max(0.0f,x);
    m_roiWindow.top    = max(0.0f,y);
    m_roiWindow.right  = m_roiWindow.left + width;
    m_roiWindow.bottom = m_roiWindow.top + height;
    }
    
//Inside OnFrameArrived:
 auto frame = sender.TryGetNextFrame();
 auto swapChainResizedToFrame = TryResizeSwapChain(frame);
 auto surfaceTexture = GetDXGIInterfaceFromObject<ID3D11Texture2D>(frame.Surface());
 m_d3dContext->CopySubresourceRegion(m_stagingTexture, 0, 0, 0, 0, surfaceTexture.get(), 0, &m_roiWindow);
 D3D11_MAPPED_SUBRESOURCE mappedSubResource;
 auto hr = m_d3dContext->Map(m_stagingTexture, 0, D3D11_MAP_READ, 0, &mappedSubResource);


<Use mappedSubResource.pData for something>

//Unmap
m_d3dContext->Unmap(m_stagingTexture, 0);
frame.Close();
```

An issue I had is that I had to make sure the width and height of the region is 32bit aligned, but that is probably because I use the data in an openGL texture. You might not have that issue.
